### DISCUSSION TITLE
reset data src

### DISCUSSION THREAD

**Discussion by pettaa123** [alexzhornyak/SCXML-tutorial#20] [Q&A] [answered]
It is possible to assign values to data's attribute expr.
Why is it not possible to manipulate data's attribute src?

**Comment by alexzhornyak** [comment, marked as answer]
It is possible to do with [\<scxml\> attribute 'binding=late'](https://alexzhornyak.github.io/SCXML-tutorial/Doc/scxml.html#late)

![image](https://user-images.githubusercontent.com/18611095/165304996-34c3c5fe-5865-4577-b95c-45782692ce37.png)

```xml
<scxml binding="late" datamodel="ecmascript" name="ScxmlShape1" version="1.0" xmlns="http://www.w3.org/2005/07/scxml">
	<state id="StateShape1">
		<datamodel>
			<data id="Var1" src="script1.js"/>
		</datamodel>
		<onentry>
			<log expr="Var1" label="Var1"/>
		</onentry>
		<transition event="step" target="StateShape2"/>
	</state>
	<state id="StateShape2">
		<datamodel>
			<data id="Var1" src="script2.js"/>
		</datamodel>
		<onentry>
			<log expr="Var1" label="Var1"/>
		</onentry>
		<transition event="step" target="FinalShape1"/>
	</state>
	<final id="FinalShape1"/>
</scxml>
```

**script1.js**
```js
"this is the 1 script"
```

**script2.js**
```js
"this is the 2 script"
```

**Output:**
> [Log] Var1: "this is the 1 script"
> [Log] Var1: "this is the 2 script"
